# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fs/test/gtest/FileSystemMocks.h
# Commit: 214c4d6bf7f9
# Full Hash: 214c4d6bf7f98a332c4ab81d325d9245ae810955
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2022-10-20 21:51:26
# Regressor Bug: 1792245
# File Overlap Count: 2
# Description:
#   Bug 1792245 - Add ErrorResult to FileSystemRequestHandler methods; r=dom-storage-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D159572
# ==============================================================================

diff -r 330c69218a93 -r 214c4d6bf7f9 dom/fs/test/gtest/FileSystemMocks.h
--- a/dom/fs/test/gtest/FileSystemMocks.h	Thu Oct 20 07:33:41 2022 +0300
+++ b/dom/fs/test/gtest/FileSystemMocks.h	Thu Oct 20 04:21:30 2022 +0000
@@ -52,54 +52,58 @@
 class MockFileSystemRequestHandler : public FileSystemRequestHandler {
  public:
   MOCK_METHOD(void, GetRootHandle,
-              (RefPtr<FileSystemManager> aManager, RefPtr<Promise> aPromise),
+              (RefPtr<FileSystemManager> aManager, RefPtr<Promise> aPromise,
+               ErrorResult& aError),
               (override));
 
   MOCK_METHOD(void, GetDirectoryHandle,
               (RefPtr<FileSystemManager> & aManager,
                const FileSystemChildMetadata& aDirectory, bool aCreate,
-               RefPtr<Promise> aPromise),
+               RefPtr<Promise> aPromise, ErrorResult& aError),
               (override));
 
   MOCK_METHOD(void, GetFileHandle,
               (RefPtr<FileSystemManager> & aManager,
                const FileSystemChildMetadata& aFile, bool aCreate,
-               RefPtr<Promise> aPromise),
+               RefPtr<Promise> aPromise, ErrorResult& aError),
               (override));
 
   MOCK_METHOD(void, GetFile,
               (RefPtr<FileSystemManager> & aManager,
-               const FileSystemEntryMetadata& aFile, RefPtr<Promise> aPromise),
+               const FileSystemEntryMetadata& aFile, RefPtr<Promise> aPromise,
+               ErrorResult& aError),
               (override));
 
   MOCK_METHOD(void, GetEntries,
               (RefPtr<FileSystemManager> & aManager, const EntryId& aDirectory,
                PageNumber aPage, RefPtr<Promise> aPromise,
-               RefPtr<FileSystemEntryMetadataArray>& aSink),
+               RefPtr<FileSystemEntryMetadataArray>& aSink,
+               ErrorResult& aError),
               (override));
 
   MOCK_METHOD(void, RemoveEntry,
               (RefPtr<FileSystemManager> & aManager,
                const FileSystemChildMetadata& aEntry, bool aRecursive,
-               RefPtr<Promise> aPromise),
+               RefPtr<Promise> aPromise, ErrorResult& aError),
               (override));
 
   MOCK_METHOD(void, MoveEntry,
               (RefPtr<FileSystemManager> & aManager, FileSystemHandle* aHandle,
                const FileSystemEntryMetadata& aEntry,
                const FileSystemChildMetadata& aNewEntry,
-               RefPtr<Promise> aPromise),
+               RefPtr<Promise> aPromise, ErrorResult& aError),
               (override));
 
   MOCK_METHOD(void, RenameEntry,
               (RefPtr<FileSystemManager> & aManager, FileSystemHandle* aHandle,
                const FileSystemEntryMetadata& aEntry, const Name& aName,
-               RefPtr<Promise> aPromise),
+               RefPtr<Promise> aPromise, ErrorResult& aError),
               (override));
 
   MOCK_METHOD(void, Resolve,
               (RefPtr<FileSystemManager> & aManager,
-               const FileSystemEntryPair& aEndpoints, RefPtr<Promise> aPromise),
+               const FileSystemEntryPair& aEndpoints, RefPtr<Promise> aPromise,
+               ErrorResult& aError),
               (override));
 };
 
@@ -144,7 +148,7 @@
   // PromiseNativeHandler implementation
 
   void ResolvedCallback(JSContext* aCx, JS::Handle<JS::Value> aValue,
-                        ErrorResult& aRv) override {
+                        ErrorResult& aError) override {
     mozilla::ScopeExit flagAsDone([isDone = mIsDone, timer = mTimer] {
       timer->Cancel();
       *isDone = true;
@@ -154,7 +158,7 @@
   }
 
   void RejectedCallback(JSContext* aCx, JS::Handle<JS::Value> aValue,
-                        ErrorResult& aRv) override {
+                        ErrorResult& aError) override {
     mozilla::ScopeExit flagAsDone([isDone = mIsDone, timer = mTimer] {
       timer->Cancel();
       *isDone = true;